### PR TITLE
update ignored_dirs in boilerplate.py and update pylint in pre-commit config to ignore boilerplate.py

### PR DESCRIPTION
**Pull request by nawazkh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- This PR will add `"_artifacts"`, `"hack/tools/bin"` folders to the list of _ignored_dirs_ in `boilerplate.py`.
- This PR also updates `pylint` config in the pre-commit framework to ignore linting to `boilerplate.py`. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
- This PR is for https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/4038

**Special notes for your reviewer**:

- This PR solves the errors I see on running pre-commit-config locally.
  - Boilerplate errors due to local files
```bash
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/_codespell.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/_version.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/__init__.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/__main__.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/tests/test_dictionary.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/tests/test_basic.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/tests/__init__.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/hack/tools/bin/codespell_dist/codespell_lib/data/__init__.py
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.26.1/manifests/.!65523!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.26.1/manifests/.!57237!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.26.1/manifests/generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.26.1/manifests/.!9501!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!66305!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!77716!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!53990!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!28694!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!11900!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!75046!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!19917!generate.sh
Boilerplate header is wrong for: /Users/nawazhussain/msftcode/cluster-api-provider-azure/_artifacts/calico/release-v3.25.1/manifests/.!87949!generate.sh
```

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nawazkh** [comment]
Apologies, I missed updating the PR title. Please take a look now.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ repos:
   rev: v2.17.2
   hooks:
   - id: pylint
+    exclude: ^hack/boilerplate/boilerplate.py$
```
Comment (nawazkh) [inline .pre-commit-config.yaml:23]: I added this to make pylint ignore the linter errors reported on updating `boilerplate.py`

**Review by mboersma — COMMENTED** [review]
/lgtm

I've run into this error a few times with the commit hook. 👍🏻

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**